### DISCUSSION TITLE
How to use repeater data to execute a REST API?

### DISCUSSION THREAD

**Discussion by Jpdsj** [Budibase/budibase#12773] [Help]
I have a repeater that brings a certain number of data according to a search I perform.

![Captura de Tela 2024-01-12 às 21 38 12](https://github.com/Budibase/budibase/assets/63756053/b8aeb491-9b9b-451f-83ba-4c9b1a6e8e69)

After fetching, I need to store this data using a REST API.

So, I need to pass this data in a structure similar to this:

"packing": [
  {
    "product": 1,
    "price": 10
  },
  {
    "product": 2,
    "price": 20
  }
]

How can I do this?

**Comment by ConorWebb96** [comment]
Hey @Jpdsj,

Have you looked into [update state](https://docs.budibase.com/docs/data-actions#update-state)? With this persisted you could pass state between pages.

Hopefully this helps.
